doc/api/errors.md

```diff
@@ -1004,6 +1004,22 @@ Used when a callback is called more then once.
 can either be fulfilled or rejected but not both at the same time. The latter
 would be possible by calling a callback more then once.
 
+<a id="ERR_NET_INVALID_LISTEN_ARG"></a>
+### ERR_NET_INVALID_LISTEN_ARG
```

Comment:
There is an ongoing debate on whether error codes should be very specific (like this case), or more general (like what @Trott is suggesting)
For now the trend is towards a clear separation between bad arguments or options, and run time errors. But minimal separation between different argument errors.
@pmatzavin IMHO, the description you've given here should be in the `net.Server.listen` API documentation, and `ERR_INVALID_ARG_TYPE` is good enough.